apcd-cms/src/apps/utils/apcd_database.py

```diff
@@ -328,13 +328,13 @@ def update_registration_entity(form, reg_id, iteration, no_entities):
     conn = None
     values = ()
     try:
+        str_end = f'{iteration}_{reg_id}'
         if not _acceptable_entity(form, iteration, reg_id):
             if iteration <= no_entities: # entity is not in form but was in original entity list -> need to delete
-                return delete_registration_entity(reg_id, form[f'ent_id_{iteration}'])
+                return delete_registration_entity(reg_id, form['ent_id_{}'.format(str_end)])
```

Comment:
When I hit the remove entity button, I get a console error. It only seems to remove the Entity 2 header, but doesn't remove the last entity. If you exit modal and come back, the entity and is still there and the Entity 2 header comes back. Is this button supposed to remove that entity from the DB for this registered submitter?
```
Uncaught TypeError: Cannot set properties of null (setting 'innerHTML')
    at HTMLButtonElement.<anonymous> 
```

Reply:
Fixed!